reporter.go

```diff
@@ -0,0 +1,302 @@
+package main
+
+import (
+	"encoding/xml"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/rainforestapp/rainforest-cli/rainforest"
+	"github.com/urfave/cli"
+)
+
+// Maximum concurrency for multithreaded HTTP requests
+const reporterConcurrency = 4
+
+// resourceAPI is part of the API connected to available resources
+type reporterAPI interface {
+	GetRunTestDetails(int, int) (*rainforest.RunTestDetails, error)
+}
+
+type reporter struct {
+	getRunDetails               func(int, *rainforest.Client) (*rainforest.RunDetails, error)
+	createOutputFile            func(string) (*os.File, error)
+	createJunitReportSchema     func(*rainforest.RunDetails, *rainforest.Client) (*jUnitReportSchema, error)
+	createJunitTestReportSchema func(int, *[]rainforest.RunTestDetails, reporterAPI) (*[]jUnitTestReportSchema, error)
+	writeJUnitReport            func(*jUnitReportSchema, *os.File) error
+}
+
+func createReport(c *cli.Context) error {
+	r := newReporter()
+	return r.createReport(c)
```

Comment:
Hmmm that's kind of a pain because when I return `cli.NewExitError`, it prints out the error message with nothing else. But Russ likes the errors in log format with a timestamp. But I'll figure it out.